src/utils_internal.rs

```diff
@@ -191,6 +195,54 @@ pub async fn build_tcp_stream(address: String) -> Result<tokio::net::TcpStream,
     Ok(stream)
 }
 
+const MSH_SERVICE: Uuid = Uuid::from_u128(0x6ba1b218_15a8_461f_9fa8_5dcae273eafd);
+
+async fn scan_peripherals(adapter: &Adapter) -> Result<Vec<Peripheral>, btleplug::Error> {
+    adapter
+        .start_scan(ScanFilter {
+            services: vec![MSH_SERVICE],
+        })
+        .await?;
+    adapter.peripherals().await
+}
+
+/// Finds a BLE radio matching a given name and running meshtastic.
+/// It searches for the 'MSH_SERVICE' running on the device.
+async fn find_ble_radio(name: String) -> Result<Peripheral, Error> {
```

Comment:
How do we export this function currently? I'm not seeing it being made public

Reply:
Sorry, I should have made this more clear. This function is not  supposed to be exposed, it's the first step of the future `build_ble_stream` function, see the screenshot.
![Screenshot_20240119_080445](https://github.com/meshtastic/rust/assets/3822689/a86d764d-707f-4b01-835e-4f2222080d93)

I'm splitting the BLE support into smaller PRs, so that it's easier to review. Otherwise it would be too much at once IMO.